x/slashing/client/rest/tx.go

```diff
@@ -1,74 +0,0 @@
-// This software is Copyright (c) 2019-2020 e-Money A/S. It is not offered under an open source license.
-//
-// Please contact partners@e-money.com for licensing related questions.
-
-package rest
-
-import (
-	"bytes"
-	"net/http"
-
-	"github.com/gorilla/mux"
-
-	"github.com/cosmos/cosmos-sdk/client/context"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/types/rest"
-	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
-	"github.com/e-money/em-ledger/x/slashing/types"
-)
-
-func registerTxRoutes(cliCtx context.CLIContext, r *mux.Router) {
-	r.HandleFunc(
-		"/slashing/validators/{validatorAddr}/unjail",
-		unjailRequestHandlerFn(cliCtx),
-	).Methods("POST")
-}
-
-// Unjail TX body
-type UnjailReq struct {
-	BaseReq rest.BaseReq `json:"base_req" yaml:"base_req"`
-}
-
-func unjailRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		vars := mux.Vars(r)
-
-		bech32validator := vars["validatorAddr"]
-
-		var req UnjailReq
-		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
-			return
-		}
-
-		req.BaseReq = req.BaseReq.Sanitize()
-		if !req.BaseReq.ValidateBasic(w) {
-			return
-		}
-
-		valAddr, err := sdk.ValAddressFromBech32(bech32validator)
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
-		}
-
-		fromAddr, err := sdk.AccAddressFromBech32(req.BaseReq.From)
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
-			return
-		}
-
-		if !bytes.Equal(fromAddr, valAddr) {
-			rest.WriteErrorResponse(w, http.StatusUnauthorized, "must use own validator address")
-			return
-		}
-
-		msg := types.NewMsgUnjail(valAddr)
-		err = msg.ValidateBasic()
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
-			return
-		}
-
-		utils.WriteGenerateStdTxResponse(w, cliCtx, req.BaseReq, []sdk.Msg{msg})
-	}
-}
```

Comment:
Noticed the same for the Market module :) WiP or not doing Rest posts?

Reply:
good question:
The `slashing` module uses the vanilla SDK REST files now See https://github.com/e-money/em-ledger/blob/dev_stargate_slashing/x/slashing/module.go#L79 instead of copying the files to your repo.

The `market` module had only REST support for queries. The `TX` file is also empty in master: https://github.com/e-money/em-ledger/blob/master/x/market/client/rest/tx.go.